### PR TITLE
feat(MWPW-137714|MWPW-137598)

### PR DESCRIPTION
**Pull request by sharath-kannan**
1)Synchronous file operations changed to Asynchronous.
2)Comment thread added.
3)Code lens provider is now class.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 node_modules
-*.vsix
\ No newline at end of file
+*.vsix
+out
```
Comment (salonijain3) [inline .gitignore:3]: add a new line

File: src/extension.ts [hunk 2]
```diff
@@ -1,15 +1,20 @@
...
 
-import * as vscode from "vscode";
 import * as fs from "fs";
 import path = require("path");
-import { getSnippet } from "./snippet";
+import * as vscode from "vscode";
 import { config } from "./config";
-import { showComments, lineChangeFlag } from "./showComments";
-import showCommentListPanel from "./webView/webViewPanel";
+import { languages } from 'vscode';
```
Comment (drashti1712) [inline src/extension.ts]: Since we are already importing * from vscode, is a separate import for languages required here?

File: src/extension.ts [hunk 3]
```diff
@@ -26,207 +31,225 @@ export class NewComment implements vscode.Comment {
...
+          continue;
+          }
+          const uri = <vscode.Uri>path || currentActiveFile || vscode.window.activeTextEditor?.document.uri;
+          const commentThread = commentController.createCommentThread(uri, new vscode.Range(lineNumber - 1, 0, lineNumber - 1, 0), []);
+          commentThread.canReply = false;
+          commentThread.collapsibleState = vscode.CommentThreadCollapsibleState.Collapsed;
+          const myNewComment = new NewComment(value, lineNumber, vscode.CommentMode.Editing,
+            { name: "" },
+            commentThread,
+            "what's a context value!"
```
Comment (drashti1712) [inline src/extension.ts]: If this is not required, could we replace it with an empty string?

File: src/extension.ts [hunk 4]
```diff
@@ -26,207 +31,225 @@ export class NewComment implements vscode.Comment {
...
-    config.currentFilePath = event.document.uri.path;
-    config.commentJSONPath = (newStr2 + ".json");
-    config.document = vscode.window.activeTextEditor?.document;
-    config.folderName = vscode.workspace.workspaceFolders !== undefined? vscode.workspace.workspaceFolders[0].name : "";
-    config.fileName = newStr2.slice(newStr2.lastIndexOf('/')+1,newStr2.length);
-    commentProvider.dispose();
-    commentProvider = showComments();
+    if (event.document.uri.scheme === 'file' && currentActiveFile?.path !== event?.document.uri.path && !event.document.uri.path.includes('commentinput-1')) {
+      currentActiveFile = event?.document.uri;
+      console.log(event.selection.active.line);
```
Comment (drashti1712) [inline src/extension.ts]: @sharath-kannan Could we remove console logs if not required?

File: src/extension.ts (hunk 3 above)
Comment (sharath-kannan) [reply src/extension.ts]: done

File: src/extension.ts (hunk 4 above)
Comment (sharath-kannan) [reply src/extension.ts]: removed.

**Review by salonijain3 — CHANGES_REQUESTED** [review]

**Pull request merged**